src/index.ts

```diff
@@ -557,10 +560,38 @@ app.after(() => {
           requireParam(req.body, "publicIdentifier");
         }
         const client = multiClient.getClient(req.body.publicIdentifier);
-        res.status(200).send<RouteMethods.PostDepositResponse>(await client.deposit(req.body));
+        const balances = await client.balance(constants.AddressZero);
+        let gas: BigNumber | undefined;
+        const gasPrice = await client.client?.ethProvider.getGasPrice();
+        if (req.body.assetId === constants.AddressZero) {
+          gas = await client.client?.ethProvider.estimateGas({
+            to: client.client.multisigAddress,
+            value: req.body.amount,
+          });
+        } else {
+          const contract = new Contract(req.body.assetId!, tokenAbi, client.client?.ethProvider);
+          gas = await contract.estimateGas.transfer(
+            client.client!.multisigAddress,
+            req.body.amount,
+          );
+        }
+        if (!gas || !gasPrice) {
+          return res.status(400).send<GenericErrorResponse>({ message: "Could not estimate gas." });
+        }
+        const totalEthRequired = gas!
+          .mul(gasPrice)
+          .add(req.body.assetId === AddressZero ? req.body.amount : 0);
+        if (BigNumber.from(balances.freeBalanceOnChain).lt(totalEthRequired)) {
+          return res.status(400).send<GenericErrorResponse>({
+            message: `Signer address balance ${balances.freeBalanceOnChain} is less than required amount ${totalEthRequired}`,
```

Comment:
```suggestion
            message: `Signer address Ether balance ${balances.freeBalanceOnChain} is less than required amount ${totalEthRequired}`,
```

Just to clarify that Ether is needed for this transaction which might not be intuitive for people with little experience with the Ethereum ecosystem and ERC20 tokens

Reply:
Very true! Thanks for the suggestion!